storage_service/locations/tests/test_dataverse.py

```diff
@@ -83,3 +106,43 @@ def test_move_to(self):
         assert 'metadata' in os.listdir(self.dest_path)
         assert 'agents.json' in os.listdir(os.path.join(self.dest_path, 'metadata'))
         assert 'dataset.json' in os.listdir(os.path.join(self.dest_path, 'metadata'))
+
+    def test_get_query_and_subtree(self):
+        """Test the function that we're using to construct parts of the
+        Dataverse query from the relative_path string stored in the Storage
+        Service.
+        """
+        query_tests = [
+            {"query": "query: drinks_1 subtree: archivematica_1/111",
+             "result": ("111", "archivematica_1", "drinks_1")},
+
+            {"query": "Query: Drinks_2 Subtree: Archivematica_2/222",
+             "result": ("222", "archivematica_2", "drinks_2")},
+
+            {"query": "query: drinks_3 subtree: archivematica_3",
+             "result": (None, "archivematica_3", "drinks_3")},
+
+            {"query": "query: drinks_4",
+             "result": (None, None, "drinks_4")},
+
+            {"query": "query: drinks_5/555",
+             "result": ("555", None, "drinks_5")},
+
+            {"query": "",
+             "result": (None, None, "*")},
+
+            {"query": "Subtree: Archivematica_7",
+             "result": (None, "archivematica_7", "*")}
+        ]
+
+        for test in query_tests:
+            # Retrieve our query our result sets.
+            query_ = test.get("query")
+            res_ = test.get("result")
+
+            # Assign values to: Dataset, Subtree, Query
+            d1, s1, q1 = self.dataverse.get_query_and_subtree(query_)
+            d2, s2, q2 = res_
+            assert d2 == d2
+            assert s1 == s2
+            assert q1 == q2
```

Comment:
Nitpick: no need for defensive `.get(` in lines 140,141. Also, lines 144-148 could just be `assert res_ == self.dataverse.get_query_and_subtree(query_)`. Bug: `assert d2 == d2` should be `assert d1 == d2`, but that would be avoided if my first suggestion were followed.